packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -49,66 +49,42 @@ export default function NavigationMenuSelector( {
 		onFinishMenuCreation
 	);
 
+	const showSelectMenus =
+		( canUserSwitchNavigation || canUserCreateNavigation ) &&
+		( navigationMenus?.length || hasClassicMenus );
+
+	if ( ! showSelectMenus ) {
+		return null;
+	}
+
 	return (
 		<>
-			<MenuGroup label={ __( 'Menus' ) }>
-				<MenuItemsChoice
-					value={ ref }
-					onSelect={ ( selectedId ) =>
-						onSelect(
-							navigationMenus.find(
-								( post ) => post.id === selectedId
-							)
-						)
-					}
-					choices={ navigationMenus.map( ( { id, title } ) => {
-						const label = decodeEntities( title.rendered );
-						return {
-							value: id,
-							label,
-							'aria-label': sprintf(
-								/* translators: %s: The name of a menu. */
-								__( "Switch to '%s'" ),
-								label
-							),
-						};
-					} ) }
-				/>
-			</MenuGroup>
+			<ExistingMenusOptions
+				showNavigationMenus={ canUserSwitchNavigation }
```

Comment:
It looks like these attributes are exactly the same as in the other place where `<ExistingMenusOptions` is used. Do they have to be configurable? What if the component just implemented all that logic?

Reply:
Yeh I was wondering about that. 

My initial thinking was extract all UI into reusable component and then folllow up with another PR creating one component to rule them all.

However I _could_ just do this in one hit if you're happy with the overhead from a PR reviewer PoV.

Reply:
I haven't forgotten about this.